modin/experimental/xgboost/xgboost_ray.py

```diff
@@ -202,28 +153,43 @@ def _set_row_partitions_to_actors(
 ):
     row_partitions_by_actors = {ip: ([], []) for ip in actors}
     if evenly_data_distribution:
-        num_actors = len(actors)
-        row_parts_last_idx = (
-            len(row_partitions) // num_actors
-            if len(row_partitions) % num_actors == 0
-            else len(row_partitions) // num_actors + 1
+        _distribute_partitions_evenly(
+            actors,
+            row_partitions_by_actors,
+            row_partitions,
+            is_partitions_have_ip=False,
         )
-        start_idx = 0
-        for ip, actor in actors.items():
-            idx_slice = (
-                slice(start_idx, start_idx + row_parts_last_idx)
-                if start_idx + row_parts_last_idx < len(row_partitions)
-                else slice(start_idx, len(row_partitions))
-            )
-            row_partitions_by_actors[ip][0].extend(row_partitions[idx_slice])
-            start_idx += row_parts_last_idx
     else:
-        partitions_ips = [ray.get(row[0]) for row in row_partitions]
-
         if data_for_aligning is None:
-            for i, row_part in enumerate(row_partitions):
-                row_partitions_by_actors[partitions_ips[i]][0].append(row_part[1])
-                row_partitions_by_actors[partitions_ips[i]][1].append(i)
+            actors_ips = list(actors.keys())
+            partitions_ips = [ray.get(row_part[0]) for row_part in row_partitions]
+
+            empty_actor_ips = []
+            for ip in actors_ips:
+                if ip not in set(partitions_ips):
+                    empty_actor_ips.append(ip)
+
+            # In case portion of nodes without data is more than 10%, data
+            # redistribution between nodes will be applied.
```

Comment:
```suggestion
            # In case portion of nodes without data is less than 10%,
            # no data redistribution between nodes will be performed.
```
so that comment applies to `then` branch instead of `else`

Reply:
fixed